packages/@react-spectrum/button/src/ActionButton.tsx

```diff
@@ -40,6 +42,7 @@ function ActionButton(props: SpectrumActionButtonProps, ref: FocusableRef<HTMLBu
       <button
         {...styleProps}
         {...buttonProps}
+        {...hoverProps}
```

Comment:
I think we should ...mergeProps(buttonProps, hoverProps). This should be done everywhere, you shouldn't need to merge with styleProps